src/cluster/index.js

```diff
@@ -177,11 +177,20 @@ module.exports = class Cluster {
    * @return {Promise}
    */
   async addMultipleTargetTopics(topics) {
+    // check if any topics need to be added, skip lock if possible
+    if (
+      this.brokerPool.metadata &&
+      this.previousTopics &&
+      topics.every(topic => this.previousTopics.has(topic))
```

Comment:
why do we need to use previousTopics here?  why not just targetTopics ?  And related, why do we need to make previousTopics an object property?

Reply:
@t-d-d , saving 'previousTopics' as an object property is needed since here we are not holding the lock, and other flow could be already inside the lock and modified 'targetTopics' before calling 'refreshMetadata', so the 'targetTopics' might be reverted on 'refreshMetadata' error. So saving the 'previousTopics' is needed as the latest topics that successfully finished the 'refreshMetadata'